### ISSUE TITLE
[BUG] Always executing ServerLogic of Base Endpoint

### ISSUE THREAD

**Issue by QuadStingray** [softwaremill/tapir#1687]
Tapir version: 0.19.1

Scala version: 2.13.7

What is the problem?
When you have an Base Endpoint with Server Logic like `/customers/1` and the Endpoint has "Sub"-Endpoints like `/customers/1/orders` always the Base Endpoint Server Logic is executed. 

Maybe you can provide code to reproduce the problem?
See code Sample

```
import akka.actor.ActorSystem
import akka.http.scaladsl.Http
import akka.http.scaladsl.server.{ Route, RouteConcatenation }
import sttp.client3._
import sttp.model.headers.WWWAuthenticateChallenge
import sttp.model.{ Method, StatusCode }
import sttp.tapir._
import sttp.tapir.model._
import sttp.tapir.server.akkahttp._

import scala.concurrent.duration.DurationInt
import scala.concurrent.{ Await, ExecutionContext, Future }

object BasicAuthenticationAkkaServer extends App with RouteConcatenation {
  implicit val ex: ExecutionContext = ExecutionContext.global

  private val baseEndpointDefinition = endpoint
    .securityIn("secret" / "user")
    .securityIn(path[String]("userId"))
    .securityIn(auth.basic[UsernamePassword](WWWAuthenticateChallenge.basic("example")))
    .serverSecurityLogic(credentials => authorize(credentials))

  val secret = baseEndpointDefinition
    .out(stringBody)
    .method(Method.GET)
    .serverLogic(username => param => Future.successful(Right(s"Hello, $username!")))

  val secret2 = baseEndpointDefinition
    .in("campaign")
    .in(path[String]("campaignId"))
    .in("hello")
    .out(stringBody)
    .method(Method.GET)
    .serverLogic(username => param => Future.successful(Right(s"Hello, $username! This is secret Route2. You would like to Check Campaign (Id: $param)!")))

  private def authorize(credentials: (String, UsernamePassword)): Future[Either[Unit, String]] = {
    Future.successful(Right(credentials._2.username + " (Id: " + credentials._1 + ")"))
  }

  val secretRoute: Route = concat(AkkaHttpServerInterpreter().toRoute(secret), AkkaHttpServerInterpreter().toRoute(secret2))

  implicit val actorSystem: ActorSystem = ActorSystem()

  val bindAndCheck = Http().newServerAt("localhost", 8080).bindFlow(secretRoute).map { _ =>
    val backend: SttpBackend[Identity, Any] = HttpURLConnectionBackend()

    val result = basicRequest.get(uri"http://localhost:8080/secret/user/1").header("Authorization", "Basic dXNlcjpzZWNyZXQ=").send(backend)
    println("Got result: " + result)
    assert(result.code == StatusCode.Ok)
    assert(result.body == Right("Hello, user (Id: 1)!"))

    val result2 = basicRequest.get(uri"http://localhost:8080/secret/user/1/campaign/12/hello").header("Authorization", "Basic dXNlcjpzZWNyZXQ=").send(backend)
    println("Got result: " + result2)
    assert(result2.code == StatusCode.Ok)
    assert(result2.body == Right("Hello, user (Id: 1)!"))
    assert(result2.body == Right("Hello, user (Id: 1)! This is secret Route2. You would like to Check Campaign (Id: 12)!"))

  }

  Await.result(bindAndCheck.transformWith(r => actorSystem.terminate().transform(_ => r)), 1.minute)

}

```

**Comment by QuadStingray** [comment]
@adamw Maybe it depends on the Usage of path Variables at the SecurityLogic? https://github.com/softwaremill/tapir/issues/1659

**Comment by adamw** [comment]
Hm yes I think there's a problem when there's no additional path defined in the "regular" inputs section - then the path-exhausted check is not done. Should be fixable :)

**Comment by adamw** [comment]
@QuadStingray I'll release a new version once sonatype is working again, currently releases are not possible

**Comment by QuadStingray** [comment]
Thanks

**Comment by adamw** [comment]
Fixed in 0.19.3
